tree.go

```diff
@@ -763,6 +744,7 @@ func (n *InternalNode) GetProofItems(keys keylist) (*ProofElements, []byte, [][]
 		if child != nil {
 			points[i] = child.Commitment()
 		} else {
```

Comment:
The next three `// TODO` comments are three branches that are still uncovered.

These are the three branches of `tree.go` that, IMHO, would be good to cover to have "complete basic" coverage of important things. I didn't handle this in this PR since I'm planning to understand more in detail theory and implementation of proof generation/verification in the short-term; so leaving covering these angles at that point.